mne/annotations.py

```diff
@@ -856,6 +863,13 @@ def events_from_annotations(raw, event_id=None, regexp=None, use_rounding=True,
 
     annotations = raw.annotations
 
+    if event_id == "auto":
+        from .io.brainvision.brainvision import RawBrainVision
```

Comment:
If we want to go that route I would put all these things in a function and register the default behaviors there.

```
event_id = get_auto_event_id(raw.__class__()) if event_id == 'auto' else event_id
```
then remove `None` from the parameters and return the default behaviour if raw does not quack properly. (which is what is happening here: https://github.com/mne-tools/mne-python/blob/master/mne/annotations.py#L778)

it would be something like this:

```py
class RawBrainVision(BaseRaw):

    @classmethod
    def auto_event_id(cls):
        pass

...

def get_auto_event_id(kls):
    foo = kls.auto_event_id if 'auto_event_id' in kls.__dict__.keys() else _Counter()
    return foo
```

(plus we should check that the parameter `event_id` is only is a dict, a callable or a string strictly equal to `'auto'`)

Reply:
Yes, this can be refactored to make it nicer, but it's imperative it gets in at .18.

Reply:
I agree that getting this done for 0.18 should be our priority for now. @massich you could also add the classmethod to the base class and return a counter function there, then you don't even have to write that if clause.